components/properties/src/names.rs

```diff
@@ -257,18 +257,55 @@ impl<T: TrieValue> PropertyParserBorrowed<'static, T> {
 
 /// Avoid monomorphizing multiple copies of this function
 fn get_strict_u16(payload: &PropertyValueNameToEnumMapV1<'_>, name: &str) -> Option<u16> {
-    // NormalizedPropertyName has no invariants so this should be free, but
-    // avoid introducing a panic regardless
-    let name = NormalizedPropertyNameStr::parse_byte_slice(name.as_bytes()).ok()?;
-    payload.map.get_copied(name)
+    payload.map.get(name).and_then(|i| i.try_into().ok())
 }
 
 /// Avoid monomorphizing multiple copies of this function
 fn get_loose_u16(payload: &PropertyValueNameToEnumMapV1<'_>, name: &str) -> Option<u16> {
-    // NormalizedPropertyName has no invariants so this should be free, but
-    // avoid introducing a panic regardless
-    let name = NormalizedPropertyNameStr::parse_byte_slice(name.as_bytes()).ok()?;
-    payload.map.get_copied_by(|p| p.cmp_loose(name))
+    fn recurse(mut cursor: ZeroTrieSimpleAsciiCursor, name: &[u8]) -> Option<usize> {
+        if cursor.is_empty() {
+            return None;
+        }
+
+        // Skip whitespace, underscore, hyphen in trie.
+        for skip in [b'\t', b'\n', b'\x0C', b'\r', b' ', 0x0B, b'_', b'-'] {
+            let mut skip_cursor = cursor.clone();
+            skip_cursor.step(skip);
+            if let Some(r) = recurse(skip_cursor, name) {
+                return Some(r);
+            }
```

Comment:
Nit: It's probably faster to check for `None` in the return value of `.step` in order to avoid the extra recursion call. In most cases the step will be `None`.